### PR TITLE
ESM-SIP(feat): First iteration system info page

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
## What changed
- Replace all the hello code with my idea code skeleton
- Added tables for all the information
- Updated the ReadMe.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
     "@babel/preset-react": "^7.13.13",
     "@babel/preset-typescript": "^7.13.0",
     "@babel/runtime": "^7.14.0",
-    "@openmrs/esm-framework": "next",
+    "@openmrs/esm-framework": "^3.2.1-pre.1155",
```
Comment (ZacButko) [inline package.json]: When working on the real repo we ask everyone to set these tags back to "next" after doing the `yarn upgrade`

File: src/systemInfo.tsx [hunk 2]
```diff
@@ -9,22 +9,15 @@
...
 import React from "react";
-import styles from "./hello.css";
-import { Greeter } from "./greeter/greeter";
-import { PatientGetter } from "./patient-getter/patient-getter";
-import { Boxes } from "./boxes/slot/boxes";
+import styles from "./systemInfo.css";
+import { Tables } from "./tables/slot/tables";
 
-const Hello: React.FC = () => {
+const SystemInformation: React.FC = () => {
```
Comment (ZacButko) [inline src/systemInfo.tsx]: This is something I have a hard time getting used to for OpenMRS project. For me I would name this file `SystemInfo.tsx` (with capital S) for a file whose main export is a React Component. However, unfortunately it is the convention of the community to have the file be kebab case like `system-info.tsx` or `system-info.component.tsx`. I really dislike this system, but that is how the community is structuring their file systems.

File: src/tables/extensions/databaseTable.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+const DatabaseTable: React.FC = () => {
+  const url = `/ws/rest/v1/systeminformation`;
+  const { data, error, isValidating } = useSWR<
+    { data: { systemInformation } },
+    Error
+  >(url, openmrsFetch);
+
+  const isLoading = !data && !error;
+
+  const headerData = [
```
Comment (ZacButko) [inline src/tables/extensions/databaseTable.tsx]: Static variables with no dependence on the component can be defined outside of the component to improve render cycles

File: src/tables/extensions/memoryTable.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+const MemoryTable: React.FC = () => {
+  const url = `/ws/rest/v1/systeminformation`;
+  const { data, error, isValidating } = useSWR<
+    { data: { systemInformation } },
+    Error
+  >(url, openmrsFetch);
+
+  const isLoading = !data && !error;
+
+  const headerData = [
```
Comment (ZacButko) [inline src/tables/extensions/memoryTable.tsx]: These table files seem to be very similar. I would look at making the component more reusable, or even its functions and constants, like `headerData` here which we already said can live outside of the main component

File: translations/en.json [hunk 5]
```diff
@@ -1,4 +1 @@
-{
-  "casualGreeting": "hey",
-  "formalGreeting": "hello"
```
Comment (ZacButko) [inline translations/en.json:3]: Don't forget to add translations back in. For example for the table headers.

**Review by ZacButko — APPROVED** [review]
Code looks good. Lets go over it next week :)

File: package.json (hunk 1 above)
Comment (hadijahkyampeire) [reply package.json]: Okay, I will do that.

File: src/systemInfo.tsx (hunk 2 above)
Comment (hadijahkyampeire) [reply src/systemInfo.tsx]: 🤔 I think different companies have different conventions, actually, at my company, we emphasize camelCase for component naming and even functions, I will use kebab case for this.

File: src/tables/extensions/databaseTable.tsx (hunk 3 above)
Comment (hadijahkyampeire) [reply src/tables/extensions/databaseTable.tsx:16]: The data is going to be different, I created a reusable component for the tables that will receive rows and headers as props, these five tables will have different headers and rowData.

File: src/tables/extensions/memoryTable.tsx (hunk 4 above)
Comment (hadijahkyampeire) [reply src/tables/extensions/memoryTable.tsx]: The data is going to be different, I created a reusable component for the tables that will receive rows and headers as props, these five tables will have different rowData.

File: translations/en.json (hunk 5 above)
Comment (hadijahkyampeire) [reply translations/en.json:3]: Sure I will put header translations

**Pull request merged**